### ISSUE TITLE
covscan: memory leak in audit log when adding entries

### ISSUE THREAD

**Issue by progier389** [389ds/389-ds-base#5647]
**Issue Description**
Covscan reports:
```
1. Defect type: RESOURCE_LEAK
19. 389-ds-base-2.2.4/ldap/servers/slapd/auditlog.c:231: leaked_storage: Variable "vals" going out of scope leaks the storage it points to.
```
And code review confirms that a charray_free is missing. 
Impact: if nsslapd-auditlog-display-attrs is not *  all added entries attributes values that are logged in audit log are leaking. 

**Package Version and Platform:**
 - Platform: RedHat
 - Package and version: 389-ds-base-2.2.4

**Steps to Reproduce**
1. Generate source package 
2. Have a covscan run using these sources
3. Look at the report about the"Resources leaks" section

**Expected results**
No leaks should be reported.

**Comment by progier389** [comment]
JIRA ticket is [IDMDS-2386]

**Comment by mreynolds389** [comment]
@progier389 - looks like this introduced a compiler warning:

```
../389-ds-base/ldap/servers/slapd/auditlog.c: In function 'add_entry_attrs':
../389-ds-base/ldap/servers/slapd/auditlog.c:257:27: warning: unused variable 'vals' [-Wunused-variable]
  257 |             Slapi_Value **vals = attr_get_present_values(entry_attr);
      |                           ^~~~
```

**Comment by progier389** [comment]
> @progier389 - looks like this introduced a compiler warning:

Indeed.  I forgot to remove that line when I moved that code to log_entry_attr !

**Comment by tbordaz** [comment]
Issue 5647 - Fix unused variable warning from previous commit (#5670)
c69f2691b..fd6b417fc  389-ds-base-2.3
4ea307115..9c61eab66  389-ds-base-2.2
43ce87bcb..7103aea66  389-ds-base-2.1
1dafcb8da..89d592856  389-ds-base-2.0
dddb14210..be7c2b829  389-ds-base-1.4.3

issue 5647 - covscan: memory leak in audit log when adding entries (#5650)
8a14402f8..db7be9fbe  389-ds-base-2.3
879cfd0b8..4ea307115  389-ds-base-2.2
b43184b0f..43ce87bcb  389-ds-base-2.1 
ed0c0feff..1dafcb8da  389-ds-base-2.0
2cea6b249..dddb14210  389-ds-base-1.4.3

**Comment by Firstyear** [comment]
@tbordaz You forgot 1.4.4 as well :)

**Comment by tbordaz** [comment]
Ooops forgot to mention
Ticket 5647 is a bug introduced with https://github.com/389ds/389-ds-base/issues/5502.
https://github.com/389ds/389-ds-base/issues/5502 is not fixed in 1.4.4, so 5647 does not apply to 1.4.4

**Comment by Firstyear** [comment]
Ahhh got it. Thanks!
